linkerd/meshtls/rustls/src/creds/store.rs

```diff
@@ -236,27 +242,6 @@ impl rustls::server::ResolvesServerCert for CertResolver {
         &self,
         hello: rustls::server::ClientHello<'_>,
     ) -> Option<Arc<rustls::sign::CertifiedKey>> {
-        let server_name = match hello.server_name() {
-            Some(name) => {
-                let name = webpki::DnsNameRef::try_from_ascii_str(name)
-                    .expect("server name must be a valid server name");
-                webpki::SubjectNameRef::DnsName(name)
-            }
-            None => {
-                debug!("no SNI -> no certificate");
-                return None;
-            }
-        };
-
-        // Verify that our certificate is valid for the given SNI name.
-        let c = self.0.cert.first()?;
-        if let Err(error) = webpki::EndEntityCert::try_from(c.as_ref())
-            .and_then(|c| c.verify_is_valid_for_subject_name(server_name))
-        {
-            debug!(%error, "Local certificate is not valid for SNI");
-            return None;
-        };
-
```

Comment:
This is removed, because if we assume that our certificate can holds its Id in either URI or DNS SAN, we cannot pose the restriction that the Server Name that is contained in the ClientHello matches the value of the SAN. This further begs the question of whether we should use a resolver at all. An alternative approach is to just set the certificates that will be used when we construct the tls config.

Reply:
Is there analogous change on the boring side?

Reply:
No, not really. In boring we set the certificate that is going to be used, rather than relying on a resolver to pick it. There is the option to do that here as well. Not sure why the resolver has been picked as an approach. Maybe to add an extra check to ensure that the cert will have the SNI from the ClientHello.

Reply:
I expect it's so that the TLS library may vary the certificates it presents based on the SNI value.